File: VMO-Back/Model/Model/ContractType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Model.Model
{
    public class ContractType
    {
        [Key]
        public string ContractTypeId { get; set; }

        [Required, MinLength(4, ErrorMessage = "Yêu cầu nhập tên hợp đồng")]
        public string Name { get; set; }
        public string ContractCode { get; set; }
        public int Term { get; set; }
        public int Status { get; set; }
        public ICollection<Contract> Contracts { get; set; }
        
    }
}

File: VMO-Frontend/src/app/model/ContractType.ts
import { ActiveStatus } from "src/share/model/ActiveStatus";
import { Contract } from './Contract';

export interface ContractType {
  contractTypeId: string;
  name: string;
  contractCode: string;
  term: number;
  status: ActiveStatus;
}

export interface ContractTypeOverview {
  contractTypeId: string;
  contractTypeName: string;
  number: number;
}
